demux/demux_mf.c

```diff
@@ -126,10 +126,10 @@ static mf_t *open_mf_pattern(void *talloc_ctx, struct demuxer *d, char *filename
 
 #if HAVE_GLOB
     if (!strchr(filename, '%')) {
-        strcpy(fname, filename);
-        if (!strchr(filename, '*'))
-            strcat(fname, "*");
-
+        if (strchr(filename, '*'))
+            strncpy(fname, filename, fname_avail - 1);
+        else
+            snprintf(fname, fname_avail, "%s*", filename);
```

Comment:
Style nit but don't we normally use braces for if/else blocks even if they're one line?